vault/router.go

```diff
@@ -522,14 +524,17 @@ func (r *Router) routeCommon(ctx context.Context, req *logical.Request, existenc
 	r.l.RLock()
 	adjustedPath := req.Path
 	mount, raw, ok := r.root.LongestPrefix(ns.Path + adjustedPath)
+	r.logger.Trace("trying to route to mount using adjusted path", "namespace", ns, "adjustedPath", adjustedPath, "mount", mount, "raw", raw, "ok", ok)
```

Comment:
Isn't this going to be a log message per request?  That seems too heavy.

Reply:
It's possible that I'm misunderstanding how the logger works but won't this be effectively a no-op unless the log level is actually set to `Trace`? I'm guessing most people don't have the log level set to `Trace` unless they're actively debugging something. And if it does get set to `Trace`, I feel like it's a reasonable expectation that you trade off performance for increased verbosity. It feels appropriate to me in this instance (as do the rest of the log lines in this file), but I wonder if there's a missing piece in my understanding here.

Reply:
Most people don't run with trace.  Some do.  Those people who do, should not suddenly post-upgrade get logging that scales based on request volume.

Reply:
I removed all the trace logging and augmented the error responses instead. My main concern here was we have identical error messages being returned for 3 different failure scenarios and it would be helpful for debugging if we were able to tell which failure scenario was being triggered.

Reply:
For sure!